### ISSUE TITLE
Pausa obligatoria; bloquear teclado en las pausas.

### ISSUE THREAD

**Issue by javidominguez** [jpavonabian/Gestor-de-Pomodoros#2]
Mi problema con los pomodoros es que no respeto las pausas. Cuando estoy muy concentrado en algo no hago caso de los avisos de pausa y sigo trabajando. A veces ni me entero. Estaría bien algo que me obligara a parar.

Prueba a añadir esto a tu clase GlobalPlugin a ver que te parece. Es algo radical pero se podría pulir un poco.

    def getScript(self, gesture):
        if self.pomodoro_thread.in_break or self.pomodoro_thread.long_break:
            return self.script_reportPomodoroStatus
        else:
            return globalPluginHandler.GlobalPlugin.getScript(self, gesture)

**Comment by jpavonabian** [comment]
A priori bloquear el teclado en las pausas no me convence. Puede ser... Peliagudo.

Quizás puliéndolo un poco para que permitiera desbloquearlo o que fuese configurable podría ser.

Un saludo.

________________________________
De: Javi Dominguez ***@***.***>
Enviado: lunes, 8 de abril de 2024 23:52
Para: jpavonabian/Gestor-de-Pomodoros ***@***.***>
Cc: Subscribed ***@***.***>
Asunto: [jpavonabian/Gestor-de-Pomodoros] Pausa obligatoria; bloquear teclado en las pausas. (Issue #2)


Mi problema con los pomodoros es que no respeto las pausas. Cuando estoy muy concentrado en algo no hago caso de los avisos de pausa y sigo trabajando. A veces ni me entero. Estaría bien algo que me obligara a parar.

Prueba a añadir esto a tu clase GlobalPlugin a ver que te parece. Es algo radical pero se podría pulir un poco.

def getScript(self, gesture):
    if self.pomodoro_thread.in_break or self.pomodoro_thread.long_break:
        return self.script_reportPomodoroStatus
    else:
        return globalPluginHandler.GlobalPlugin.getScript(self, gesture)

**Comment by javidominguez** [comment]
Sí, claro, habría que darle una vuelta. Se podría dejar pasar una combinación de teclas que desbloquee el teclado si hace falta. Por ejemplo, los mismos atajos que sirven para pausar o detener el Pomodoro. También un ajuste de configuración para que active esto solo el usuario que lo quiera.
Pero bueno, era solo una idea, si no te convence pues nada. Aunque sí que creo que, para mí, le haría falta algo que llame la atención más que un ui.messaje qué es fácil de pasar por alto.

**Comment by jpavonabian** [comment]
Le estoy dando una vuelta, me han dicho que se han perdido pausas justo por eso. No me parece mala la idea.


________________________________
De: Javi Dominguez ***@***.***>
Enviado: martes, 9 de abril de 2024 9:47
Para: jpavonabian/Gestor-de-Pomodoros ***@***.***>
Cc: Jesús Pavón Abián ***@***.***>; Comment ***@***.***>
Asunto: Re: [jpavonabian/Gestor-de-Pomodoros] Pausa obligatoria; bloquear teclado en las pausas. (Issue #2)


Sí, claro, habría que darle una vuelta. Se podría dejar pasar una combinación de teclas que desbloquee el teclado si hace falta. Por ejemplo, los mismos atajos que sirven para pausar o detener el Pomodoro. También un ajuste de configuración para que active esto solo el usuario que lo quiera.
Pero bueno, era solo una idea, si no te convence pues nada. Aunque sí que creo que, para mí, le haría falta algo que llame la atención más que un ui.messaje qué es fácil de pasar por alto.

**Comment by javidominguez** [comment]
Lo de bloquear el teclado quizá era un poco radical. Se me ha ocurrido otra idea. ¿Qué te parece si, estando en un periodo de pausa detecta que el usuario sigue tecleando dé otro aviso? No demasiado insistente, a lo mejor cada 10, 15 o 20 segundos si se han detectado pulsaciones de teclas.

**Comment by jpavonabian** [comment]
No estaría mal. Incluso bloquear el teclado siempre y cuando puedas
controlarlo con un checkbox y desbloquearlo si lo necesitas tampoco estaría
mal.
Voy a estudiarme wx a ver si doy con la forma de añadir una categoría en el
menú de opciones de NVDA para estas cosas.
